src/dispatchers/browserContextDispatcher.ts

```diff
@@ -63,6 +64,25 @@ export class BrowserContextDispatcher extends Dispatcher<BrowserContext, channel
         this._dispatchEvent('serviceWorker', { worker: new WorkerDispatcher(this._scope, serviceWorker)});
       context.on(CRBrowserContext.CREvents.ServiceWorker, serviceWorker => this._dispatchEvent('serviceWorker', { worker: new WorkerDispatcher(this._scope, serviceWorker) }));
     }
+    context.on(BrowserContext.Events.Request, (request: Request) => this._dispatchEvent('request', {
+      request: RequestDispatcher.from(this._scope, request),
+      page: request.frame()._page
```

Comment:
This is the place where `page` might not be initialized yet (as in not finished the setup), so sending it over protocol would be an issue. That's why I suggest to add a method like `page.initializedOrNull()` that will return the page itself after the `reportAsNew` was called.

Reply:
Even after calling reportAsNew, it fails to initialize the page. Can you help here?